### PR TITLE
Add Merge Sort

### PR DESCRIPTION
**Pull request by ruddelbenito**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -69,7 +72,38 @@ function splitArray(nums) {
 }
 
 function combine(arrays) {
```
Comment (TresTres) [inline main.js]: can rename this to merge

File: main.js [hunk 2]
```diff
@@ -69,7 +72,38 @@ function splitArray(nums) {
 }
 
 function combine(arrays) {
-    result = [];
+    // if one of the two arrays are empty, return the other array
+    if (arrays[0].length === 0) {
```
Comment (TresTres) [inline main.js]: If you're going to use something multiple times, it's better to assign it to a variable.  
`arrays[0]` is accurate but it means remembering that `arrays` contains multiple arrays, and it only contains 2 arrays.  Using `let leftArr = arrays[0]` will make your code easier to follow

File: main.js [hunk 3]
```diff
@@ -69,7 +72,38 @@ function splitArray(nums) {
...
+    if (arrays[1].length === 0) {
+        return arrays[0];
+    }
+
+    let result = [];
+    let leftPointer = 0;
+    let rightPointer = 0;
+
+    while (leftPointer < arrays[0].length && rightPointer < arrays[1].length) {
+        if (arrays[0][leftPointer] < arrays[1][rightPointer]) {
```
Comment (TresTres) [inline main.js]: Take a look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift and see if this changes anything

File: main.js [hunk 4]
```diff
@@ -69,7 +72,38 @@ function splitArray(nums) {
...
+        else {
+            result.push(arrays[1][rightPointer]);
+            rightPointer++;
+        }
+    }
+
+    // if after one array is empty, the other array still has content, dump the rest of
+    // the non-empty array into result
+    if (leftPointer < arrays[0].length) {
+        result = result.concat(arrays[0].slice(leftPointer, arrays[0].length));
```
Comment (TresTres) [inline main.js]: I think you could use `...` spread operator here too.

File: main.js (hunk 1 above)
Comment (TresTres) [reply main.js]: and rename the sort function from `merge` to `mergeSort`

**Review by TresTres — CHANGES_REQUESTED** [review]

File: main.js (hunk 1 above)
Comment (ruddelbenito) [reply main.js]: Addressed

File: main.js (hunk 2 above)
Comment (ruddelbenito) [reply main.js]: Addressed

File: main.js (hunk 3 above)
Comment (ruddelbenito) [reply main.js]: Addressed

File: main.js (hunk 4 above)
Comment (ruddelbenito) [reply main.js]: Addressed

**Review by TresTres — APPROVED** [review]

**Pull request merged**